### PR TITLE
canbus: rename zephyr,can-primary chosen property to zephyr,canbus

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Rename the Zephyr chosen property for specifying the default CAN bus controller from "zephyr,can-primary" to "zephyr,canbus".

The "zephyr,can-primary" property name was selected in antipation of adding support for redundant CAN networks, which we have yet to add. Meanwhile, the "primary" term causes confusion for non-redundant CAN bus configurations (and the "can" term doesn't match the name of the Zephyr CAN bus subsystem).

The CAN in Automation (CiA) 302-6, which deals with CANopen network redundancy, uses the terms "default interface" and "redundant interface". If/when we add support for redundant CAN networks, the "zephyr,canbus" chosen property can be supplemented with a "zephyr,canbus-redundant" chosen property.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

### CODE REVIEW

**Comment by GwenStouthuysen** [comment]
The assumption here is that a potential second CAN interface is used as back-up for the main device.
I have processes that handle 3 separate CAN networks at the same time.

**Comment by henrikbrixandersen** [comment]
> The assumption here is that a potential second CAN interface is used as back-up for the main device.
> I have processes that handle 3 separate CAN networks at the same time.

Actually, no such assumption is made.

The `zephyr` prefix in the `chosen` properties is for Zephyr system-wide settings. Zephyr by itself does not have any applications with support for more than one CAN bus controller (neither for redundancy or other use cases). Zephyr by itself does however have the need for determining which CAN controller to use for simple, uni-interface CAN-enabled applications and subsystems. See https://docs.zephyrproject.org/latest/guides/dts/intro.html#aliases-and-chosen-nodes for further information on aliases and chosen nodes.

I do not foresee us introducing Zephyr in-tree code supporting e.g. three CAN controllers since that is very domain specific (I do, however, foresee us adding CANopen support with support for redundant networks, hence the comment in the commit message).

Out-of-tree applications, on the other hand, are free to define whatever `chosen` or `aliases` properties they need (but they probably should not be using the `zephyr` prefix for this). An out-of-tree application can e.g. specify following disregarding this change:

```
/ {
     aliases {
             can-primary = &can1;
             can-secondary = &can2;
             can-tertiary = &can0;
     };
};
```

Or, if that's more fitting:

```
/ {
     aliases {
             can-upstream = &can1;
             can-downstream = &can2;
     };
};
```

The main point here, is that no Zephyr _in-tree_ functionality has support for multi-homed CAN, and that if and when we do introduce that, the `chosen` nodes likely need to have more domain-specific names than `zephyr,can-primary`, `zephyr,can-secondary` etc.

File: include/devicetree/zephyr.h [hunk 1]
```diff
@@ -64,9 +64,9 @@
 	DT_LABEL(DT_CHOSEN(zephyr_flash_controller))
 #endif
 
-#if DT_NODE_HAS_PROP(DT_CHOSEN(zephyr_can_primary), label)
-#define DT_CHOSEN_ZEPHYR_CAN_PRIMARY_LABEL \
-	DT_LABEL(DT_CHOSEN(zephyr_can_primary))
+#if DT_NODE_HAS_PROP(DT_CHOSEN(zephyr_canbus), label)
+#define DT_CHOSEN_ZEPHYR_CANBUS_LABEL \
+	DT_LABEL(DT_CHOSEN(zephyr_canbus))
```
Comment (martinjaeger) [inline include/devicetree/zephyr.h:69]: I'm wondering why we need these macro definitions at all. They look like a leftover from something around Zephyr 2.1 where the `DT_CHOSEN(node)` macros didn't exist yet. Can't we just replace with the direct `DT_*` calls?

Comment (henrikbrixandersen) [reply include/devicetree/zephyr.h:69]: I agree, but I would like to leave that change to a separate PR.

**Review by martinjaeger — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Review by alexanderwachter — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
The buildkite failure is due to timeouts (builds taking longer than 3h) due to the many boards changed in this PR. How can I work-around that?

**Comment by martinjaeger** [comment]
> The buildkite failure is due to timeouts (builds taking longer than 3h) due to the many boards changed in this PR. How can I work-around that?

You can put DNM label and preliminarily increase (e.g. double) the timeout in a separate commit:
https://github.com/zephyrproject-rtos/zephyr/blob/8b773481e974ec952755e5cad1bd49604c6bb2a8/.buildkite/pipeline.yml#L7-L8

If everything builds, remove the commit again.

**Comment by henrikbrixandersen** [comment]
Buildkite passed with extended timeouts, which have been removed again. Ready for merging.

**Comment by alexanderwachter** [comment]
@cfriedt ping for merging :-)

**Comment by cfriedt** [comment]
> @cfriedt ping for merging :-)

I lack those super powers, but I'll ping the powers that be.

**Comment by alexanderwachter** [comment]
There comes the power 😆

**Pull request merged**